platform/windows/detect.py

```diff
@@ -66,7 +59,7 @@ def can_build():
         # Cross-compiling with MinGW-w64 (old MinGW32 is not supported)
         prefix = os.getenv("MINGW_PREFIX", "")
 
-        if try_cmd("gcc --version", prefix, "") or try_cmd("clang --version", prefix, ""):
+        if get_mingw_tool("gcc", prefix, "") or get_mingw_tool("clang", prefix, ""):
```

Comment:
Maybe `arch` could also default to `""` so we don't need to specify these for non-arch specific binaries?